### PR TITLE
change debug attach asserts about OOM to failfasts

### PR DESCRIPTION
**Pull request by MikeHolman**

### CODE REVIEW

File: lib/Runtime/Debug/DebugContext.cpp [hunk 1]
```diff
@@ -142,11 +142,7 @@ namespace Js
         });
         END_TRANSLATE_OOM_TO_HRESULT(hr);
 
-        if (hr != S_OK)
-        {
-            Assert(FALSE);
-            return hr;
-        }
+        DEBUGGER_ATTACHDETACH_FATAL_ERROR_IF_FAILED(hr);
```
Comment (akroshg) [inline lib/Runtime/Debug/DebugContext.cpp]: why failfast when it is OOM? Why not replace with Assert(hr==S_OK || hr==E_OUTOFMEMORY)?

Comment (agarwal-sandeep) [inline lib/Runtime/Debug/DebugContext.cpp]: Yes we should return OOM and let the caller which have fail fast handle it

Comment (MikeHolman) [inline lib/Runtime/Debug/DebugContext.cpp]: Apparently Paul is removing this function anyway, so I'll wait for that to merge.

**Comment by MikeHolman** [comment]
@dotnet-bot test Ubuntu ubuntu_linux_release_static

**Comment by akroshg** [comment]
:shipit:

**Comment by MikeHolman** [comment]
@dotnet-bot test Ubuntu ubuntu_linux_release

**Comment by MikeHolman** [comment]
@dotnet-bot test Ubuntu ubuntu_linux_release_static

**Pull request merged**